pkg/db/interface.go

```diff
@@ -105,10 +150,17 @@ func NewWithSQLConn(db *sql.DB) VizierDBInterface {
 }
 
 func New() VizierDBInterface {
+	timeToWait := time.Duration(30) * time.Second
+	err := waitForDbReady(timeToWait)
```

Comment:
readiness probe has its own merit but I think it is enough to try to connect the mysql in a loop here.
The extra function just adds unnecessary complexity.

Reply:
When the pod for vizier-db literally not ready or failing, we'd better to wait for it being ready, perhaps infinitely (though we should configure redinessProbe for vizier-core as well in that case). I set it to 30 sec since normally DB seems to get ready in 30 sec if it's properly configured, while Pod Restart due to mis-configuration or something like that is okay. OTOH, when the pod literally is ready to serve, I think we don't need retrying at all in that case, though retrying to connect with its timeout + interval set to some reasonable value might be okay.

Simple loop as you commented seems okay, while I think this PR diff is small enough. Just a minor con is that vizier-db's log would be dirtied. I can replace https://github.com/kubeflow/katib/pull/261/commits/a0d4d063b53596d39e7480532d59c922a0f27cee shortly.